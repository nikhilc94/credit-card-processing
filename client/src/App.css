.app-container {
  margin: 20px;
}

h2 {
  font-weight: 500;
}

h3 {
  font-weight: 400;
}
