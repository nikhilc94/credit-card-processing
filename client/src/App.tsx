import React, { useState } from "react";

import "./App.css";
import Form from "./components/Form";
import Table from "./components/Table";

const App: React.FC = () => {
  const [cardDetails, setCardDetails] = useState([]);

  return (
    <div className="app-container">
      <h2>Credit Card System</h2>
      <h3>Add</h3>
      <Form setCardDetails={setCardDetails} />
      <h3>Existing Cards</h3>
      <Table cardDetails={cardDetails} setCardDetails={setCardDetails} />
    </div>
  );
};

export default App;
