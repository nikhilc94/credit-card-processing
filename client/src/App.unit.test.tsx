import React from "react";
import { shallow } from "enzyme";
import { createSerializer } from "enzyme-to-json";

import App from "./App";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("App component", () => {
  it("renders successfully", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
