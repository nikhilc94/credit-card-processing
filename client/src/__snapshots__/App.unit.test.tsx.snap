// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders successfully 1`] = `
<div
  className="app-container"
>
  <h2>
    Credit Card System
  </h2>
  <h3>
    Add
  </h3>
  <Form
    setCardDetails={[Function]}
  />
  <h3>
    Existing Cards
  </h3>
  <Table
    cardDetails={Array []}
    setCardDetails={[Function]}
  />
</div>
`;
